src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -68,48 +69,101 @@ public IList<ISourceReference> GetSources()
+            {
                 var afterCompileContext = new AfterCompileContext
                 {
                     ProjectContext = CompilationContext.ProjectContext,
                     Compilation = CompilationContext.Compilation,
                     AssemblyStream = assemblyStream,
-                    SymbolStream = pdbStream,
-                    Diagnostics = new List<Diagnostic>(diagnostics)
+                    SymbolStream = pdbStream
                 };
 
-                foreach (var m in CompilationContext.Modules)
+                EmitResult emitResult = null;
+
+                if (!string.IsNullOrEmpty(assemblyName.CultureName))
                 {
-                    m.AfterCompile(afterCompileContext);
+                    var resourcesForCulture = ResourcesForCulture.GetResourcesForCulture(assemblyName.CultureName ?? string.Empty, CompilationContext.Resources);
+                    if (resourcesForCulture == null)
+                    {
+                        return null;
+                    }
+                    afterCompileContext.SymbolStream = null;
+                    emitResult = EmitResourceAssembly(assemblyName, resourcesForCulture, afterCompileContext);
+                }
+                else
+                {
+                    var resourcesForCulture = ResourcesForCulture.GetResourcesForCulture(assemblyName.CultureName ?? string.Empty, CompilationContext.Resources);
+                    if (resourcesForCulture == null)
+                    {
+                        // No resources is fine for a main assembly
+                        resourcesForCulture = Enumerable.Empty<ResourceDescriptor>();
+                    }
+                    var resources = resourcesForCulture
+                        .Select(res => new ResourceDescription(res.Name, res.StreamFactory, isPublic: true));
+
+                    Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
+
+                    var sw = Stopwatch.StartNew();
+
+                    if (_supportsPdbGeneration.Value)
+                    {
+                        emitResult = CompilationContext.Compilation.Emit(assemblyStream, pdbStream: pdbStream, manifestResources: resources);
+                    }
+                    else
+                    {
+                        Logger.TraceWarning("PDB generation is not supported on this platform");
+                        emitResult = CompilationContext.Compilation.Emit(assemblyStream, manifestResources: resources);
+                    }
+
+                    afterCompileContext.Diagnostics = CompilationContext.Diagnostics.Concat(emitResult.Diagnostics).ToList();
+
+                    sw.Stop();
+
+                    Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
+                }
+
+                if (!Path.GetExtension(assemblyName.Name).Contains("resources"))
```

Comment:
Do we have to compute this from the path again? Isn't there another indicator if we're compiling a resource assembly other than the name?